18F/identity-idp
# Extract session JavaScript package

## 🛠 Summary of changes

This pull request extracts logic reused between `session-timeout-ping.ts` and `document-capture.ts` packs related to session behavior to a new JavaScript package, `@18f/identity-session`.

Rationale:

1. Shared logic for common session API interaction
2. Better testability
3. Support upcoming work for session modal custom element as part of this package
4. Reduce scope of session-timeout-ping pack toward eventual removal
5. Improves refactorability of API endpoint by consolidating, supporting #7966

## 📜 Testing Plan

- `yarn mocha app/javascript/packages/session-timeout/index.spec.ts`
- Ensure that session timeout modal appears as expected
- Ensure that clicking "Keep me signed in" extends session as expected
- Ensure that document capture step progression (i.e. IPP) extends session timeout
